internal/hns/hnsendpoint.go

```diff
@@ -94,6 +94,29 @@ func GetHNSEndpointByName(endpointName string) (*HNSEndpoint, error) {
 	return nil, EndpointNotFoundError{EndpointName: endpointName}
 }
 
+type endpointAttachInfo struct {
+	SharedContainers json.RawMessage `json:",omitempty"`
+	VirtualMachine   json.RawMessage `json:",omitempty"`
+}
+
+func (endpoint *HNSEndpoint) IsAttached(vID string) (bool, error) {
+	attachInfo := endpointAttachInfo{}
+	err := hnsCall("GET", "/endpoints/"+endpoint.Id, "", &attachInfo)
+
+	// Return true allows us to just return the err if this is true
+	if err != nil {
+		return true, err
+	}
+
+	if strings.Contains(string(attachInfo.SharedContainers), vID) ||
```

Comment:
SharedContainers should be sufficient for v1 Argon

There is no shared containers for v1 Xenon
Why are you checking VirtualMachine ?

Reply:
For completeness I guess, I can remove if you want.

Reply:
@madhanrm fixed, take a look, I want to revendor the plugins repo w/ this.

Reply:
ugh, actually we should probably put them both to lower.

Reply:
Yes please

Reply:
Done